libs/contrib/Data/String/Parser.idr

```diff
@@ -209,22 +209,28 @@ covering
 takeWhile : Monad m => (Char -> Bool) -> ParseT m String
 takeWhile f = pack <$> many (satisfy f)
 
+||| Similar to `takeWhile` but fails if the resulting string is empty.
+export
+covering
+takeWhile1 : Monad m => (Char -> Bool) -> ParseT m String
+takeWhile1 f = pack <$> some (satisfy f)
+
 ||| Parses one or more space characters
```

Comment:
Similarly here we want *one* or more spaces.
You can always define `spaces?` as `optional spaces`

Reply:
Yes, and that is exactly what the `many` -> `some` change is about.

Reply:
Oh sorry, I will never not get confused by these two.